sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -22,17 +27,199 @@ class TableCheckpointStore:
+            dic[u'etag'] = entity.metadata.get('etag')
+            dic[u'last_modified_time'] = entity.metadata.get('timestamp')
+            ownershiplist.append(dic)
+        return ownershiplist
+
+    def list_checkpoints(self, fully_qualified_namespace, eventhub_name, consumer_group):
+        """List the updated checkpoints from the storage blob.
+        :param str fully_qualified_namespace: The fully qualified namespace that the Event Hub belongs to.
+         The format is like "<namespace>.servicebus.windows.net".
+        :param str eventhub_name: The name of the specific Event Hub the checkpoints are associated with, relative to
+         the Event Hubs namespace that contains it.
+        :param str consumer_group: The name of the consumer group the checkpoints are associated with.
+        :rtype: Iterable[Dict[str,Any]], Iterable of dictionaries containing partition checkpoint information:
+                - `fully_qualified_namespace` (str): The fully qualified namespace that the Event Hub belongs to.
+                  The format is like "<namespace>.servicebus.windows.net".
+                - `eventhub_name` (str): The name of the specific Event Hub the checkpoints are associated with,
+                  relative to the Event Hubs namespace that contains it.
+                - `consumer_group` (str): The name of the consumer group the checkpoints are associated with.
+                - `partition_id` (str): The partition ID which the checkpoint is created for.
+                - `sequence_number` (int): The sequence number of the :class:`EventData<azure.eventhub.EventData>`.
+                - `offset` (str): The offset of the :class:`EventData<azure.eventhub.EventData>`.
+        """
+        partition_key = "{} {} {} Checkpoint".format(fully_qualified_namespace,
+        eventhub_name, consumer_group)
+        my_filter = "PartitionKey eq '"+ partition_key + "'"
+        entities = self.table_client.query_entities(my_filter)
+        checkpointslist = []
+        for entity in entities:
+            dic = {}
+            dic[u'fully_qualified_namespace'] = fully_qualified_namespace
+            dic[u'eventhub_name'] = eventhub_name
+            dic[u'consumer_group'] = consumer_group
+            dic[u'partition_id'] = entity[u'RowKey']
+            dic[u'sequence_number'] = entity[u'sequencenumber']
+            dic[u'offset'] = entity[u'offset']
+            checkpointslist.append(dic)
+        return checkpointslist
 
-    def list_checkpoints(self, namespace, eventhub, consumergroup, **kwargs):
-        pass
+    def update_checkpoint(self, checkpoint):
+        """Updates the checkpoint using the given information for the offset, associated partition and
+        consumer group in the storage table.
+        Note: If you plan to implement a custom checkpoint store with the intention of running between
+        cross-language EventHubs SDKs, it is recommended to persist the offset value as an string.
```

Comment:
I took another look at the CheckpointStore that this class should extend:
https://github.com/Azure/azure-sdk-for-python/blob/82281311c5dc080afb7206040eb8df5cd1d0e9ca/sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/checkpoint_store.py#L66-L67

It looks like the comment is that the offset should be persisted as an integer to retain compatibility with other languages. So, I think you'd want to store the offset as an int in Tables, but convert it back to a string when giving it to the user.

@yunhaoling does that seem right to you? I'm curious why it's a string at all in Python though it's too late to change that now.